[Profiler] Add Test for Clear on Fork

sraikund16 opened the pull request:
Summary: Tests Fix Clear On Fork by forking a process after a profile has already been done. Afterwards we check that all the PID/TID are as expected.

Test Plan: Ran buck2 test 'fbcode//mode/dev' fbcode//caffe2/test:profiler -- --exact 'caffe2/test:profiler - test_forked_process (profiler.test_profiler.TestProfiler)'

Differential Revision: D63992036

sanrise reviewed (approved):
test looks good! thanks

aaronenyeshi reviewed (approved):
LGTM. Please take a look at the unit test failure.

sraikund16 committed:
[Profiler] Clear TLS on Init and Add Test (#137511)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/137511

Tests Fix Clear On Fork by forking a process after a profile has already been done. Afterwards we check that all the PID/TID are as expected.

Test Plan: Ran buck2 test 'fbcode//mode/dev' fbcode//caffe2/test:profiler -- --exact 'caffe2/test:profiler - test_forked_process (profiler.test_profiler.TestProfiler)'

Reviewed By: aaronenyeshi, sanrise

Differential Revision: D63992036
